File: 290-word-pattern/290-word-pattern.cpp
class Solution {
public:
    void splitString(string s, vector<string> &v){
	
        string temp = "";
        for(int i=0;i<s.length();++i){

            if(s[i]==' '){
                v.push_back(temp);
                temp = "";
            }
            else{
                temp.push_back(s[i]);
            }

        }
        v.push_back(temp);

    }
    
    bool wordPattern(string pattern, string s) {
        int n = pattern.length();
        vector<string> arr;
        splitString(s, arr);
        int l = arr.size();
        if (n != l) return false;
        map<char, string> map1;
        map<string, char> map2;
        for (int i=0;i<n;i++) {
            if (map1.count(pattern[i]) == 0) map1[pattern[i]] = arr[i];
            else if (map1[pattern[i]] != arr[i]) return false;
            if (map2.count(arr[i]) == 0) map2[arr[i]] = pattern[i];
            else if (map2[arr[i]] != pattern[i]) return false;
        }
        return true;
    }
};

File: 290-word-pattern/290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        arr = s.split(' ')
        n = len(arr)
        l = len(pattern)
        
        if (n != l):
            return False
        map1 = {}
        map2 = {}
        for i in range(n):
            if (arr[i] not in map1):
                map1[arr[i]] = pattern[i]
            else:
                if (map1[arr[i]] != pattern[i]):
                    return False
            if (pattern[i] not in map2):
                map2[pattern[i]] = arr[i]
            else:
                if (map2[pattern[i]] != arr[i]):
                    return False
        
        return True
        